main.py

```diff
@@ -80,61 +83,49 @@ def getrssi(self): #電波強度
     for k in sorted(devices.keys()):
         print(time.time(),devices[k].getrpid(),devices[k].getrssi())
         print("-----------------------------------------------------")
-    joineduserpoplist = []
-    for k,v in joinedusers.items():
-        if k in devices.keys():
-            v.rsttime()
-            devices.pop(k)
-        else:
-            v.dectime()
-            if v.remainingtime == 0:
-                joineduserpoplist.append(k)
-    #delete user
-    for k in joineduserpoplist:
-        joinedusers.pop(k)
 
-    queueduserpoplist = []
-    for k,v in queuedusers.items():
-        if k in devices.keys():
-            v.dectime()
-            devices.pop(k)
-            if v.remainingtime == 0:
-                v.rsttime()
-                joinedusers[k] = v
-                queueduserpoplist.append(k)
-        else:
-            v.inctime()
-            if v.remainingtime == 20*1+20:
-                queueduserpoplist.append(k)
-    for k in queueduserpoplist:
-        queuedusers.pop(k)
+    for k in (joinedusers.keys & devices.keys): #キーの積集合
+        devices.pop(k) # ここでpopしていいのか自信ない
+        joinedusers[k].rsttime()
+    for k in (joinedusers.keys - devices.keys): #キーの差集合
+        v=joinedusers[k]
+        v.dectime()
+        if v.remainingtime == 0:
+            joinedusers.pop(k)
+
+    for k in (queuedusers.keys & devices.keys): #キーの積集合
+        v=queuedusers[k]
+        v.dectime()
+        devices.pop(k)
+        if v.remainingtime == 0:
+            v.rsttime()
+            joinedusers[k] = v
+            queueduserpoplist.append(k)
+    for k in (queuedusers.keys - devices.keys): #キーの差集合
+        v.inctime()
```

Comment:
v is not defined. use "v=queuedusers[k]"
No attribute error can be caused.
AttributeError: 'Device' object has no attribute 'inctime'